### DISCUSSION TITLE
[GB] Flashing a cart with 39SF010A

### DISCUSSION THREAD

**Discussion by p0ojisan** [sanni/cartreader#1021] [Need help]
Hi, I'm now building a DIY kit from orange fm gb.
https://orangegb.com/orangefm/diy.html

This DIY cart uses SST 39SF010A flash memory chip on it. I tried all of flash menu in cartreader, but failed to write with an error message:
```
Flash ID: FFFF
Unknown flashrom
```

Someone can tell which flash menu I should choose? Or doesn't cartreader support this kind flash memory?

**Comment by StanMOI** [comment]
This Flash are not supported
[https://github.com/sanni/cartreader/wiki/Supported-Systems](https://github.com/sanni/cartreader/wiki/Supported-Systems)

**Reply by p0ojisan** [reply]
I got it. Will try other flasher.

Thanks!

**Reply by StanMOI** [reply]
All you need is programmer like « T56 » and adapter for TSOP-32
https://a.aliexpress.com/_Exii0oV

**Reply by p0ojisan** [reply]
T56 programmer looks pretty expensive. I will try other flash tool recommended in the kit guidance.

Thank you, anyway.

**Comment by sanni** [comment]
Could you check if the 7th pin on the flashchip connects to the third pin on the edge connector?
![orange](https://github.com/user-attachments/assets/1b71ba9d-159f-4bd7-ae1d-216e095de375)

**Reply by p0ojisan** [reply]
Yes, they are connected. I've confirmed all the pins. Result is like this:
![orange_gb_pinout](https://github.com/user-attachments/assets/e2f04ec4-1689-4e10-b652-cb84710bda40)

And also here, luckily I have some picutres before soldering.
<img src="https://github.com/user-attachments/assets/87209eff-145e-4c9f-b087-145955c43735" width=100>
<img src="https://github.com/user-attachments/assets/ae7684ec-4f2b-4441-9a2a-42d1fc25a808" width=100>

**Reply by p0ojisan** [reply]
Let me note that I bought a clone of GBxCart RW after this and it successfully flashed to this cart.

If OSCR also supports this type of flash memory, would be helpful for others.

**Reply by sanni** [reply]
Added the Orange FM 39SF cart to the GB repro menu:

![IMG_4098](https://github.com/user-attachments/assets/8fb97640-0ca4-4299-ad85-b7068f497345)

**Reply by Chomemel** [reply]
Hi Sanni,
Is it possible to add 39SFxxx to the SNES repro code and the flashrom (but I think they share the same code, right?) please? I have snes cartridges with 39SF040.

**Reply by p0ojisan** [reply]
> Added the Orange FM 39SF cart to the GB repro menu:
> 
> ![IMG_4098](https://private-user-images.githubusercontent.com/19169494/367693787-8fb97640-0ca4-4299-ad85-b7068f497345.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1NzgxODcsIm5iZiI6MTcyNjU3Nzg4NywicGF0aCI6Ii8xOTE2OTQ5NC8zNjc2OTM3ODctOGZiOTc2NDAtMGNhNC00Mjk5LWFkODUtYjcwNjhmNDk3MzQ1LkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE3VDEyNTgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjMTgzODJlMjRmODBiMDk2ZTI4NThlY2E0N2Q4Zjk5YTkwZjVlMTEyZmQ4ZDY1NzgwOGY0OWFmNTQ2YzU0NWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.D-HQo2SvjRicP8tAlUEDqnR8z8Zne1-H-WZp_naGwPI)

Thank you so much, Sanni.
I confirmed it worked successfully in HW3 as well.
<img src="https://github.com/user-attachments/assets/81b053cf-448b-4796-83b9-607ee18381c0" width=200>
